scripts/ci/check_compliance.py

```diff
@@ -561,6 +561,36 @@ def parse_kconfig(self, filename="Kconfig", hwm=None):
             # Clean up the temporary directory
             shutil.rmtree(kconfiglib_dir)
 
+    def get_logging_syms(self, kconf):
+        # Returns a set() with the names of the Kconfig symbols generated with
+        # logging template in samples/tests folders. The Kconfig symbols doesn't
+        # include `CONFIG_` and for each module declared 4 Kconfig symbols are
+        # created:
+        #   - *_LOG_LEVEL
+        #   - *_LOG_LEVEL_DBG
+        #   - *_LOG_LEVEL_ERR
+        #   - *_LOG_LEVEL_INF
+        #   - *_LOG_LEVEL_WRN
+
+        # Warning: Needs to work with both --perl-regexp and the 're' module.
+        regex = r"^\s*(?:module\s*=\s*)([A-Z0-9_]+)\s*(?:#|$)"
+
+        # Grep samples/ and tests/ for symbol definitions
+        grep_stdout = git("grep", "-I", "-h", "--perl-regexp", regex, "--",
+                          ":samples", ":tests", cwd=ZEPHYR_BASE)
+
+        names = re.findall(regex, grep_stdout, re.MULTILINE)
```

Comment:
would not it be simpler to add an extra condition to the check in `check_no_undef_outside_kconfig` to not flag configs that match a certain suffix (the ones below) in additions to the ones in `self.UNDEF_KCONFIG_WHITELIST`? Then you won't have to do a tree wide git grep.

Reply:
I am not sure to understand, for example you want to ignore *all* symbols like `*_LOG_LEVEL_*`? I am not sure it's great to do that

Reply:
yeah, I'm a bit scared of the regexp messing up some use case and thinking a plain suffix match may be more predictable, but hey it should be alright let's go with it